src/main/java/com/projectkorra/rpg/modules/randomavatar/avatar/AvatarManager.java

```diff
@@ -21,15 +22,15 @@ public class AvatarManager {
     private static final FileConfiguration defaultConfig = ConfigManager.defaultConfig.get();
     private static final String CONFIG_PATH = "Modules.RandomAvatar.";
 
-    private static HashMap<UUID, Avatar> CURRENT_AVATARS;
-    private static HashMap<UUID, PreviousAvatar> PREVIOUS_AVATARS;
+    private static HashMap<UUID, AvatarEntity> CURRENT_AVATARS;
+    private static HashMap<UUID, AvatarEntity> PREVIOUS_AVATARS;
 
     public AvatarManager() {
```

Comment:
As discussed in the discord, rather than working in parallel it's better practice to provide feedback on the existing avatarmanager and/or propose changes/rewrites to it vs creating a brand new one. This draws into questions of authorship and is not good for facilitating collaboration. From a git perspective it's also easier to track the history this way as well. You can create future Pull Requests that make changes to the AvatarManager in specific and if all the changes you wish to see are solid then you'll still have the same end result but this way the conversation is open and the changes are done ethically without conflict.

Reply:
Also both files have seen updates in this PR which was not clear or indicated in the PR description. One of these sets of updates is redundant and should be reverted with the removal of the unwanted/unneeded file.